src/commonMain/kotlin/kotools/types/collection/NotEmptyList.kt

```diff
@@ -54,6 +39,30 @@ public fun <E> Collection<E>.toNotEmptyList(): Result<NotEmptyList<E>> =
         }
         ?: Result.failure(EmptyCollectionException)
 
+/** Representation of lists that contain at least one element of type [E]. */
+@JvmInline
+@Serializable(NotEmptyListSerializer::class)
+@SinceKotoolsTypes("4.0")
+public value class NotEmptyList<out E> private constructor(
+    private val elements: List<E>
+) : NotEmptyCollection<E> {
+    override val head: E get() = elements.first()
+    override val tail: NotEmptyList<E>?
+        get() = elements.drop(1)
+            .toNotEmptyList()
+            .getOrNull()
+
+    internal constructor(head: E, tail: NotEmptyList<E>? = null) : this(
+        tail?.run { listOf(head) + elements }
+            ?: listOf(head)
+    )
+
+    /** Returns all elements of this list as a [List] of type [E]. */
+    public fun toList(): List<E> = elements
+
+    override fun toString(): String = "$elements"
```

Comment:
We should improve the documentation of this operation by adding usage examples.

Reply:
Done by improving the documentation of the `NotEmptyCollection.toString` function (see the commit b9be1a8153bd6b89a5207eb8fa01b97d31dc55f2).